app/src/features/apiClient/APIClientManager/APIClientUtils.ts

```diff
@@ -0,0 +1,138 @@
+import { EnvironmentVariables } from "backend/environment/types";
+import { RQAPI } from "../types";
+import { requestWorkerFunction, responseWorkerFunction } from "./workerScripts";
+
+interface WorkerMessage {
+  type: "HANDLE_ENVIRONMENT_CHANGES" | "COMPLETE" | "ERROR";
+  payload?: any;
+}
+
+const createWorkerCode = (fn: Function) => {
+  return `
+    self.onmessage = ${fn.toString()};
+  `;
+};
+
+const createWorker = (workerFn: Function) => {
+  const workerCode = createWorkerCode(workerFn);
+  const blob = new Blob([workerCode], { type: "application/javascript" });
+  return new Worker(URL.createObjectURL(blob));
+};
+
+const handleEnvironmentChanges = async (
+  environmentManager: any,
+  payload: {
+    variablesToSet: EnvironmentVariables;
+    variablesToRemove: string[];
+    currentVariables: EnvironmentVariables;
```

Comment:
Why are we passing `currentVariables`? Should set and remove be an upsert or remove operation (bulk update) and rest of the variables should stay as it is